### PR TITLE
Fixes MAISTRA-1775: updated lua-jit

### PR DESCRIPTION
**Pull request by dmitri-d**
Port of ffa2f17902ee7a378a9ff3166bba375751cf465b
dependencies: bump LuaJIT to 2.1 branch HEAD @ e9af1ab. (#13474)

LuaJIT 2.1.0-beta3 has the following CVEs, which don't appear
super critical for correctly functioning Lua code but prudence dictates
we should bump anyway:

- CVE-2020-15890: LuaJit through 2.1.0-beta3 has an out-of-bounds read
  because __gc handler frame traversal is mishandled.

- CVE-2020-24372: LuaJIT through 2.1.0-beta3 has an out-of-bounds read
  in lj_err_run in lj_err.c.

There is no release version beyond 2.1.0-beta3, so using HEAD of 2.1
branch.

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by dmitri-d** [comment]
This is the same as https://github.com/maistra/envoy/pull/76, but after cert-related test failures were fixed.

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -109,9 +109,9 @@ REPOSITORY_LOCATIONS = dict(
         urls = ["https://github.com/grpc/grpc/archive/v1.22.1.tar.gz"],
     ),
     com_github_luajit_luajit = dict(
-        sha256 = "409f7fe570d3c16558e594421c47bdd130238323c9d6fd6c83dedd2aaeb082a8",
-        strip_prefix = "LuaJIT-2.1.0-beta3",
-        urls = ["https://github.com/LuaJIT/LuaJIT/archive/v2.1.0-beta3.tar.gz"],
+        sha256 = "20a159c38a98ecdb6368e8d655343b6036622a29a1621da9dc303f7ed9bf37f3",
+	strip_prefix = "LuaJIT-1d8b747c161db457e032a023ebbff511f5de5ec2",
```
Comment (oschaaf) [inline bazel/repository_locations.bzl:113]: Note sure how strict we need to be with whitespace but this seems to be indented differently

File: test/test_common/BUILD [hunk 2]
```diff
@@ -108,6 +108,7 @@ envoy_cc_test_library(
         "//source/common/filesystem:directory_lib",
         "//source/common/filesystem:filesystem_lib",
         "//source/common/http:header_map_lib",
+	"//source/common/http:header_utility_lib",
```
Comment (oschaaf) [inline test/test_common/BUILD:111]: Ditto

**Review by oschaaf — APPROVED** [review]
Two whitespace nits, that aside lgtm

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (dmitri-d) [reply bazel/repository_locations.bzl]: done

File: test/test_common/BUILD (hunk 2 above)
Comment (dmitri-d) [reply test/test_common/BUILD]: done

**Review by oschaaf — APPROVED** [review]

**Pull request merged**